docs/DeveloperGuide.md

```diff
@@ -1,3 +1,1140 @@
+which will be elaborated on in the section [[Common
+Classes]{.ul}](#common-classes)
+
+### 
+
+### **UI component**
+
+1.  Each class has a specified UI class that serves the specific need
+    > for the individual class
+
+2.  There is a Main UI Class that shares all common UI functionalities
+    > amongst the classes. This will be elaborated in the Common Classes
+    > component
+
+> (Insert UI UML diagrams)
+
+### **Logic component**
+
+1.  This component would be the backbone of our program, where the
+    > functions of each individual class are crafted and contribute to
+    > the program.
+
+2.  Command is passed through the Parser.java class which determines
+    > what actions to perform. The actions to be performed are methods
+    > called in the classes stated below.
+
+[Actions:]{.ul}
+
+1.  In the menu of Doctor's Appointment, command of "add D12345 Alex M
+    > 21012021".
+
+2.  Input is passed along to the Parser.java class where the keyword
+    > "add" is detected.
+
+3.  addAppointment() method in the AppointmentActions.java class is
+    > called and adds the details into the list
+
+[Classes:]{.ul}
+
+1.  AppointmentActions
+
+2.  DrugActions
+
+3.  NurseScheduleActions
+
+4.  PatientList
+
+5.  StaffList
+
+### **Object Class component**
+
+This component is to distinguish the separate object that has been
+instantiated to hold the details of our target users. An object class
+allows for a structure that helps store details from user input in the
+format that we want to be saved.
+
+[Objects:]{.ul}
+
+1.  DoctorAppointment.java
+
+2.  Drug.java
+
+3.  Patient.java
+
+4.  NurseSchedule.java
+
+5.  Staff.java
+
+### 
+
+### **Storage component**
+
+The difference classes for storages are separated into their respective
+architecture for easier management.
+
+> [Methods:]{.ul}
+
+1.  writeToFile() - Calling upon this method whenever there is change to
+    > the data array list.
+
+2.  createFile() - Called during the start of the program when
+    > loadFile() is called and returns an error, meaning that there are
+    > no text files in storage. Thus, this method is called to create a
+    > new text file for data storage.
+
+3.  loadFile() - Called during the start of the program, to initialise
+    > the array list and load previously saved data.
+
+[Classes:]{.ul}
+
+1.  DoctorAppointmentStorage
+
+2.  DrugStorage
+
+3.  NurseScheduleStorage
+
+4.  PatientStorage
+
+5.  StaffStorage
+
+### **Common classes**
+
+1.  Main UI Class
+
+2.  Main Storage Class
+
+ 
+
+**Implementation**
+------------------
+
+This section describes how the features are being implemented. The scope
+will include the Doctor Appointments / Drugs / Nurse Schedules /
+Patients / Staff related features.
+
+Common features that can be identified across the scope include these
+distinct functionalities: Adding, Deleting and Listing. For Nurse
+Schedules/ Patients / Staff related features, there is an additional
+search functionality.
+
```

Comment:
For the features description, I think sample inputs and outputs can be included for users to be understand how the function would work.